service/forward_service.cc

```diff
@@ -0,0 +1,176 @@
+#include "service/forward_service.hh"
+
+#include <boost/range/algorithm/remove_if.hpp>
+#include <seastar/core/future-util.hh>
+#include <seastar/core/coroutine.hh>
+
+#include "schema_registry.hh"
+#include "database.hh"
+#include "query-ranges-to-vnodes.hh"
+#include "locator/abstract_replication_strategy.hh"
+#include "gms/gossiper.hh"
+#include "log.hh"
+
+#include "service/pager/query_pagers.hh"
+#include "cql3/selection/selectable.hh"
+#include "cql3/selection/selectable-expr.hh"
+#include "cql3/selection/selection.hh"
+#include "cql3/selection/raw_selector.hh"
+#include "cql3/cql_config.hh"
+#include "cql3/column_identifier.hh"
+#include "cql3/query_options.hh"
+#include "cql3/result_set.hh"
+#include "cql_serialization_format.hh"
```

Comment:
Why does the forward service need all these cql3 includes? Seems like a layering violation, we should have cql3 -> forward -> storage_proxy layering.

Reply:
?

Reply:
As you wrote below, we need these includes to reuse cql3 aggregation code.
Executing a `forward_request` on remote node mimics the original way of executing aggregation select statements - it creates `pager` and `resutl_set_builder`. To make a pager instance, `cql3::selection::selection` is needed. Due to the fact that selection cannot be serialized (and sent in `forward_request`), it's mocked inside the `forward_service::execute` method using `mock_selection` function.